specification/attestation/Attestation/models.tsp

```diff
@@ -0,0 +1,926 @@
+   * is used, for instance, to choose among a set of keys within a JWK Set
+   * during key rollover.  The structure of the "kid" value is
+   * unspecified.  When "kid" values are used within a JWK Set, different
+   * keys within the JWK Set SHOULD use distinct "kid" values.  (One
+   * example in which different keys might use the same "kid" value is if
+   * they have different "kty" (key type) values but are considered to be
+   * equivalent alternatives by the application using them.)  The "kid"
+   * value is a case-sensitive string.
+   */
+  kid?: string;
+
+  /**
+   * The "kty" (key type) parameter identifies the cryptographic algorithm
+   * family used with the key, such as "RSA" or "EC". "kty" values should
+   * either be registered in the IANA "JSON Web Key Types" registry
+   * established by [JWA] or be a value that contains a Collision-
+   * Resistant Name.  The "kty" value is a case-sensitive string.
+   */
+  kty: string;
+
+  /**
+   * RSA modulus, in Base64
+   */
+  n?: string;
+
+  /**
+   * RSA secret prime
+   */
+  p?: string;
+
+  /**
+   * RSA secret prime, with p < q
+   */
+  q?: string;
+
+  /**
+   * RSA Private Key Parameter
+   */
+  qi?: string;
+
+  /**
+   * Use ("public key use") identifies the intended use of
+   * the public key. The "use" parameter is employed to indicate whether
+   * a public key is used for encrypting data or verifying the signature
+   * on data. Values are commonly "sig" (signature) or "enc" (encryption).
+   */
+  use?: string;
+
+  /**
+   * X coordinate for the Elliptic Curve point
+   */
+  x?: string;
+
+  /**
+   * The "x5c" (X.509 certificate chain) parameter contains a chain of one
+   * or more PKIX certificates [RFC5280].  The certificate chain is
+   * represented as a JSON array of certificate value strings.  Each
+   * string in the array is a base64-encoded (Section 4 of [RFC4648] --
+   * not base64url-encoded) DER [ITU.X690.1994] PKIX certificate value.
+   * The PKIX certificate containing the key value MUST be the first
+   * certificate.
+   */
+  x5c?: string[];
+
+  /**
+   * Y coordinate for the Elliptic Curve point
+   */
+  y?: string;
+}
+
+/**
+ * The response to the OpenID metadata description document API
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "backwards compatibility"
+model OpenIDConfigurationResponse {
+  /**
+   * Types supported in the OpenID metadata API
+   */
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "Renamed in client.tsp for SDK casing"
+  response_types_supported?: string[];
+
+  /**
+   * List of the supported signing algorithms
+   */
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "Renamed in client.tsp for SDK casing"
+  id_token_signing_alg_values_supported?: string[];
+
+  /**
+   * Revocation endpoint
+   */
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "Renamed in client.tsp for SDK casing"
+  revocation_endpoint?: string;
```

Comment:
We can avoid the need for all of the casing style suppressions if you name the properties with appropriate tsp casing and use the encoded name decorator. This has 2 benefits, 1. reduce the amount renames in the client.tsp file, making it easier to maintain and read, 2. Avoid unnecessary suppressions